remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingServer.java

```diff
@@ -250,6 +261,8 @@ public void run(Timeout timeout) {
      */
     protected ChannelPipeline configChannel(SocketChannel ch) {
         return ch.pipeline()
+            .addLast(defaultEventExecutorGroup, HA_PROXY_DECODER, new HAProxyMessageDecoder())
+            .addLast(defaultEventExecutorGroup, HA_PROXY_HANDLER, new HAProxyMessageHandler())
```

Comment:
Before that, a handshake handler is needed to determine whether it is haProxyProtocol or not, and to decide whether haProxyProtocol protocol decoder and handler are required. This ensures compatibility regardless of whether it is haProxyProtocol or not.